pkg/kubectl/cmd/util/diff/diffoptions.go

```diff
@@ -0,0 +1,177 @@
+	if err != nil {
+		return err
+	}
+
+	o.Namespace, o.EnforceNamespace, err = f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+	o.Codec = runtime.NewCodec(f.JSONEncoder(), f.Decoder(true))
+	o.ext = "." + o.Output
+	o.Mapper = mapper
+	o.Typer = typer
+
+	r := resource.NewBuilder(o.Mapper, f.CategoryExpander(), o.Typer, resource.ClientMapperFunc(f.UnstructuredClientForMapping), unstructured.UnstructuredJSONScheme).
+		NamespaceParam(o.Namespace).DefaultNamespace().
+		FilenameParam(o.EnforceNamespace, &o.FilenameOptions).
+		Flatten().
+		Do()
+	err = r.Err()
+	if err != nil {
+		return err
+	}
+
+	err = r.Visit(func(info *resource.Info, err error) error {
+		if err != nil {
+			return err
+		}
+
+		o.DiffBufferModified, err = runtime.Encode(o.Codec, info.VersionedObject)
+		if err != nil {
+			return err
+		}
+
+		// Verify the object exists in the cluster
+		if err := info.Get(); err != nil {
+			if apierrors.IsNotFound(err) {
+				return err
+			} else {
+				return cmdutil.AddSourceToErr(fmt.Sprintf("retrieving current configuration of:\n%v\nfrom server for:", info), info.Source, err)
+			}
+		}
+
+		o.DiffBufferOriginal, err = kubectl.GetOriginalConfiguration(info.Mapping, info.Object)
+		if err != nil {
+			return cmdutil.AddSourceToErr(fmt.Sprintf("retrieving current configuration of:\n%v\nfrom server for:", info), info.Source, err)
+		}
+		if o.DiffBufferOriginal == nil {
+			return cmdutil.UsageErrorf(cmd, "no last-applied-configuration annotation found on resource: %s, to create the annotation, run the command with --create-annotation", info.Name)
+		}
+
+		//decode and delete the annotations fields
+		targetObj, err := runtime.Decode(unstructured.UnstructuredJSONScheme, o.DiffBufferOriginal)
+		if err != nil {
+			return nil
+		}
+		delete(targetObj.(runtime.Unstructured).UnstructuredContent()["metadata"].(map[string]interface{}), "annotations")
+
+		o.DiffBufferOriginal, err = runtime.Encode(o.Codec, targetObj)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (o *DiffOptions) Validate(f cmdutil.Factory, cmd *cobra.Command) error {
+	return nil
+}
+
+func (o *DiffOptions) RunDiffLastApplied(f cmdutil.Factory, cmd *cobra.Command) error {
+	var err error
+	o.DiffBufferModified, err = FormatConvert(o.DiffBufferModified, o.Output)
+	if err != nil {
+		return err
+	}
+	o.DiffBufferOriginal, err = FormatConvert(o.DiffBufferOriginal, o.Output)
+	if err != nil {
+		return err
+	}
+	diff := cmdutil.NewDefaultCmdTool(cmdutil.DiffCmd, f.DiffEnvs())
+
+	tmpFileSlice := []cmdutil.TempFile{
+		//be careful with the order of DiffBufferModified and DiffBufferOriginal
+		{Prefix: fmt.Sprintf("%s-diff-last-applied-original-", filepath.Base(os.Args[0])), Suffix: o.ext, Buffer: bytes.NewReader(o.DiffBufferOriginal)},
+		{Prefix: fmt.Sprintf("%s-diff-last-applied-modified-", filepath.Base(os.Args[0])), Suffix: o.ext, Buffer: bytes.NewReader(o.DiffBufferModified)},
```

Comment:
I'm not gonna change that, because openshift/origin could reuse those code, and hard code will cause problem, see similar issue here: https://github.com/kubernetes/kubernetes/pull/48750